card.go

```diff
@@ -1,49 +1,105 @@
 package goker
 
+import "fmt"
+
 type suit int8
-type rank int8
 
 const (
+	// Spade - Constant representing the suit ♠
 	Spade suit = iota
+	// Heart - Constant representing the suit ♥
 	Heart
+	// Diamond - Constant representing the suit ♦
 	Diamond
+	// Club - Constant representing the suit ♣
 	Club
 )
 
+type rank int8
+
 const (
+	// Two - Constant representing the rank 2
 	Two rank = iota + 2
+	// Three - Constant representing the rank 3
 	Three
+	// Four - Constant representing the rank 4
 	Four
+	// Five - Constant representing the rank 5
 	Five
+	// Six - Constant representing the rank 6
 	Six
+	// Seven - Constant representing the rank 7
 	Seven
+	// Eight - Constant representing the rank 8
 	Eight
+	// Nine - Constant representing the rank 9
 	Nine
+	// Ten - Constant representing the rank 10
 	Ten
+	// Jack - Constant representing the rank Jack
 	Jack
+	// Queen - Constant representing the rank Queen
 	Queen
+	// King - Constant representing the rank King
 	King
+	// Ace - Constant representing the rank Ace
 	Ace
 )
 
-type PlayingCard interface {
+// Card - Represents a playing card, with a suit and rank
+type Card interface {
 	Rank() rank
 	Suit() suit
+	String() string
 }
 
 type card struct {
 	rank rank
 	suit suit
 }
 
-func (c card) Rank() rank {
+func (c *card) Rank() rank {
 	return c.rank
 }
 
-func (c card) Suit() suit {
+func (c *card) Suit() suit {
```

Comment:
Is that really what you want? A card will be <= 64 bits; a pointer to a card will also be 64 bits.  And then you have to pay for the indirection.  Why not just keep it as it was?

Reply:
Yeah, reading other discussion on this PR, sounds like this interface will end up going away entirely. Interesting that a pointer would take more space than the actual struct; I hadn't realized that was possible.